src/app/plugins/pluginsearch.plugin.ts

```diff
@@ -0,0 +1,55 @@
+import { Plugin } from '../../common/plugin';
+import { ChannelType, IContainer, IMessage, IPlugin } from '../../common/types';
+
+export class PluginSearchPlugin extends Plugin {
+  public name: string = 'Plugin Search';
+  public description: string = 'Search our plugins database';
+  public usage: string = 'search <phrase>';
+  public pluginAlias = ['search', 'searchplugin', 'searchplugins'];
+  public permission: ChannelType = ChannelType.Bot;
+  public commandPattern: RegExp = /[^]+/;
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public async execute(message: IMessage, args: string[]) {
+    const query = args.join(' ');
+
+    // For every plugin, evaluate it's match
+    const results: string[] = [];
+    Object.entries(this.container.pluginService.plugins).forEach(([pluginName, plugin]) => {
```

Comment:
If somebody has a more jslike way of populating this results array please chime in. Maybe a reduce? I couldn't get the syntax out.

Reply:
i wouldn't sweat it but something like
```
Object.entries(this.container.pluginService.plugins).reduce((results, [pluginName, plugin]) => {
  if (
     plugin.permission === ChannelType.Admin || 
     plugin.permission === ChannelType.Staff ||
     !this._grep(plugin, query)
   ) {
          return results;
  }

  results.push(pluginName);
  return results;
}, []);
```

Reply:
Sick! Thank you :D